firmware/libraries/versavis/src/ADIS16445.cpp

```diff
@@ -373,20 +373,31 @@ float ADIS16445::tempScale(int16_t sensorData) {
 // Method to update the sensor data without any validity checks (may result in
 // spikes).
 ///////////////////////////////////////////////////////////////////////////////////////////////
-void ADIS16445::updateData() { sensor_data_ = sensorReadAll(); }
+bool ADIS16445::updateData() {
+  sensor_data_ = sensorReadAll();
+  return true;
+}
 
 /////////////////////////////////////////////////////////////////////////////////////////////
 // Method to update the internally stored sensor data recusivelly by checking
 // the validity.
 ///////////////////////////////////////////////////////////////////////////////////////////////
-void ADIS16445::updateDataRecursive(unsigned int depth) {
+bool ADIS16445::updateDataRecursive(const unsigned int depth, bool success) {
   sensor_data_ = sensorReadAll();
   // Assuming that 1) temperature changes much slower than IMU sampling rate, 2)
   // all other measurements are correct if TEMP_OUT is correct. It may be
   // necessary to filter outliers out by using moving average filter before
   // publishing IMU topic (i.e., versavis_imu_receiver.cpp)
-  if (sensor_data_[7] != regRead(TEMP_OUT) &&
-      depth < max_recursive_update_depth_) {
-    updateDataRecursive(depth + 1);
+  if (sensor_data_[7] != regRead(TEMP_OUT)) {
```

Comment:
In addition to my other comment here: why are you doing this recursively in the first place? Wouldn't be an iterative approach more appropriate?